### ISSUE TITLE
Some questions about training stage

### ISSUE THREAD

**Issue by LiuZhe6** [Picsart-AI-Research/SeMask-Segmentation#1]
Thank you very much for your pioneering work. I have a few questions about the training phase and hope to get your answers.
1. What are the models and numbers of graphics cards used in training？
2. What is the training duration for different capacity models?
3. I'm not sure whether the network can work, If I set the batch size to 1, it's due to the particularity of the task I study.

Thank you. I'm looking forward to your reply.

**Comment by praeclarumjj3** [comment]
@LiuZhe6 Thanks for the praise. Here are the answers to your questions:

1. The number of GPUs used during training depends on the backbone variant of the Swin Transformer. We train all our models on Nvidia RTX A6000 GPUs. For the tiny, small, and base variants, we used 2 GPUs, and for the large variant, we used 4 GPUs. You may find the setting for each model on the [last line](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/354cf289434e76685bc6b33ebc6d37843d556e56/SeMask-FPN/configs/semask_swin/ade20k/semfpn_semask_swin_tiny_patch4_window7_512x512_80k_ade20k.py#L37) of their [config](https://github.com/Picsart-AI-Research/SeMask-Segmentation#1-results) files. You may find all the models under the [Results](https://github.com/Picsart-AI-Research/SeMask-Segmentation#1-results) section.

2. The training duration of the models with the Semantic-FPN decoder is 80k iterations. We mention the same on [Page 6](https://arxiv.org/pdf/2112.12782.pdf) under **Training Settings** in **Section 4.2**. You may also find the [training schedule](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/354cf289434e76685bc6b33ebc6d37843d556e56/SeMask-FPN/configs/semask_swin/ade20k/semfpn_semask_swin_tiny_patch4_window7_512x512_80k_ade20k.py#L3) in the corresponding config files. For MaskFormer, FaPN, and Mask2Former based models, we train for 160k iterations same as the baselines.

3. I assume you are talking about the batch size during inference. We evaluate the models with `batch size = 1`.

**Comment by LiuZhe6** [comment]
Thank you for your patience. I think your answer has completely answered my doubts

**Comment by LiuZhe6** [comment]
Hi, happy new year!

After reading your code, I'm a little uncertain.
After the loss function is calculated, you added an accuracy calculation item into the “loss” variable. Look at [here](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/2379a857c2620e4150c7e5c944159f6693cbf2d7/SeMask-FPN/mmseg/models/decode_heads/semask_decode_head.py#L247).
Whether this will lead to an incorrect loss in the final calculation because all items in the loss variable will be added in the summation process. [Here](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/2379a857c2620e4150c7e5c944159f6693cbf2d7/SeMask-FPN/mmseg/models/segmentors/base.py#L216).

I don't know if I'm looking for the right position, because the MM framework dazzles me.

**Comment by praeclarumjj3** [comment]
Hi @LiuZhe6, happy new year to you too!

The `loss[acc_seg]` term is in the original MMSegmentation codebase. It calculates and prints the per-pixel accuracy after every certain number of iterations.

Regarding your doubts with loss term, the [line you pointed out](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/2379a857c2620e4150c7e5c944159f6693cbf2d7/SeMask-FPN/mmseg/models/segmentors/base.py#L216) adds the terms which have `loss` in their [keys](https://github.com/Picsart-AI-Research/SeMask-Segmentation/blob/2379a857c2620e4150c7e5c944159f6693cbf2d7/SeMask-FPN/mmseg/models/segmentors/base.py#L217), so the loss is summed correctly.

**Comment by LiuZhe6** [comment]
> 

Oh! 
It's my fault that I didn't see it in this line.

Thank you!

**Comment by LiuZhe6** [comment]
If I want to change the number of GPUs and `samples_per_gpu`,  do I need to modify the learning rate?

**Comment by praeclarumjj3** [comment]
No, you don't need to change the lr.
